samples/sensor/qdec/sample.yaml

```diff
@@ -0,0 +1,8 @@
+sample:
+  description: Usage quadrature decoder sensor
+  name: qdec_sensor
+tests:
+  sample.sensor.qdec_sensor:
+    tags: sensors
+    platform_allow: nucleo_f401re
+    build_only: true
```

Comment:
with build_olny: true you won't be able to test it with twister at all, even if you have the required setup. I assume the sample needs an extra sensor connected to a platform. If so, you can use fixture keyword, and then twister will only allow to execute this sample if a proper setup (with a fixture) is defined https://docs.zephyrproject.org/latest/develop/test/twister.html#fixtures If not, it will just build the sample on a given platform.
From my understanding we require samples to also define a test which can validate them, if possible. If the sample provides some output, based on which a user is able to validate if it worked (e.g. "sensor active, readout XXX") this can be automated in twister, using console: harness and defining required regexes